src/components/Form/CreateActivationForm/CreateActivationForm.tsx

```diff
@@ -26,32 +24,41 @@ const CreateActivationForm = () => {
   const [loading, setLoading] = useState(false);
   const { triggerTooltip } = useTooltip();
 
-  const throwErrorTooltip = () => {
-    triggerTooltip({
-      severity: Severity.DANGER,
-      text:
-        "Errore durante la creazione dell'operatore, controllare i dati e riprovare"
-    });
-  };
-
-  const createActivation = async (organization: OrganizationWithReferents) =>
-    await tryCatch(
-      () => Api.AttributeAuthority.upsertOrganization(organization),
-      toError
-    )
-      .chain(chainAxios)
-      .map(response => response.data)
-      .fold(
-        () => {
-          setLoading(false);
-          throwErrorTooltip();
-        },
-        () => {
-          setLoading(false);
+  const createActivation = async (organization: OrganizationWithReferents) => {
+    try {
+      const response = await Api.AttributeAuthority.upsertOrganization(
+        organization
+      );
+      if (response instanceof AxiosError) {
+        if (
+          response.response?.status === 400 &&
+          response.response?.data === "CANNOT_BIND_MORE_THAN_TEN_ORGANIZATIONS"
+        ) {
+          triggerTooltip({
+            severity: Severity.WARNING,
+            text:
+              "Gli utenti indicati possono gestire un numero massimo di 10 operatori. Controlla e riprova."
+          });
+        } else {
+          throw new Error();
+        }
+      } else {
+        if (response.status === 200) {
           history.push(ADMIN_PANEL_ACCESSI);
+        } else {
+          throw new Error();
         }
-      )
-      .run();
+      }
+    } catch (error) {
+      triggerTooltip({
+        severity: Severity.DANGER,
+        text:
+          "Errore durante la creazione dell'operatore, controllare i dati e riprovare"
+      });
+    } finally {
+      setLoading(false);
+    }
+  };
```

Comment:
What do you think to create a more abstract and reusable handler function that avoid the boilerplate?

I suggest to implement something like this:

```typescript
import { AxiosError } from "axios";

const decodeAxiosResponse = <T>(response: any): T => {
  if (response instanceof AxiosError) {
    if (response.response?.status >= 200 && response.response?.status <= 299) {
      return response.response.data;
    } else {
      // It throws an error if the response is not considered a success response
      throw response;
    }
  }
  // If it isn't AxiosError, it assumes that it's a normal response
  return response;
};
```

And then the block of code of the `createActivation` function become:

```typescript
const createActivation = async (organization: OrganizationWithReferents) => {
  try {
    const response = await Api.AttributeAuthority.upsertOrganization(organization);
    const data = decodeAxiosResponse(response);
  
    // When it reaches this point, it means that it's a success response
    history.push(ADMIN_PANEL_ACCESSI);
  } catch (error) {
    if (
      error?.response?.status === 400 &&
      error?.response?.data === "CANNOT_BIND_MORE_THAN_TEN_ORGANIZATIONS"
    ) {
      triggerTooltip({
        severity: Severity.WARNING,
        text:
          "Gli utenti indicati possono gestire un numero massimo di 10 operatori. Controlla e riprova.",
      });
    } else {
      triggerTooltip({
        severity: Severity.DANGER,
        text:
          "Errore durante la creazione dell'operatore, controllare i dati e riprovare.",
      });
    }
  }
}

```

Reply:
i pushed two alternatives inspired by your snippet

Reply:
we decided in call the solution direction